tests/cephadm/test_package_validation.py

```diff
@@ -0,0 +1,162 @@
+    "ansible-collection-ansible-posix",
+    "ansible-collection-community-general",
+]
+
+rpm_client_packages = [
+    "ceph-common",
+    "ceph-fuse",
+    "cephfs-top",
+]
+
+
+def validate_rpm_version(node, package_list, version_config):
+    """Validate installed rpm package version
+
+    Args:
+        node (ceph.ceph.CephNode): Node name
+        package_list (list): List of packages
+        version_config (dict): List of packages and package version
+    """
+    # Define regex to get version
+    rpm_regex = r"Version\s+:\s+(\d+\.\d+)"
+
+    # Check rpm package version using yum info
+    for pkg in package_list:
+        out = Package(node).info(pkg)
+        version = re.search(rpm_regex, out).group(1)
+        if version < version_config.get(pkg):
+            raise PackageError(
+                f"Expected version for {pkg} is {version_config.get(pkg)}; found {version}"
+            )
+        else:
+            log.info(f"{pkg} installed correctly with version {version}")
+
+
+def validate_image_version(node, image_dict, version_config):
+    """Validate image version
+
+    Args:
+        node (ceph.ceph.CephNode): Node name
+        image_dict (dict): List of images
+        version_config (dict): List of packages and package version
+    """
+    # Check container image version using podman inspect
+    for key, image in image_dict.items():
+        version = Container(node).inspect(image, format="'{{.Config.Labels.version}}'")[
+            0
+        ]
+        if version < version_config.get(key):
+            raise PackageError(
+                f"Expected version for {image} is {version_config.get(image)}; found {version}"
+            )
+        else:
+            log.info(f"{image} installed correctly with version {version}")
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify package version validation
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer and client node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    client = ceph_cluster.get_nodes(role="client")[0]
+    log.info(f"Check type: {type(installer)}")
+
+    # Check if requied packages are installed
+    for pkg in installed_packages:
+        if not Rpm(installer).query(pkg):
+            raise PackageError(f"Required package {pkg} is not installed")
+
+    # Get config details
+    config = kwargs.get("config")
+    base_url = config.get("base_url")
+    cloud_type = config.get("cloud_type")
+
+    # Enable ceph tools repo on installer and client
+    base_url = get_custom_repo_url(base_url, cloud_type)
+    for node in {installer, client}:
+        if not Package(node).add_repo(base_url):
+            raise ReposError(f"Failed to enable {base_url} repo")
+
+    # Get the Tools repo file name
+    out = installer.get_dir_list(dir_path=YUM_REPO_DIR)
+    file_name = [repo for repo in out if "Tools" in repo][0]
+
+    # Get expected package versions from config
```

Comment:
Move this to start of the method

Reply:
Again this placement is correct after we enable tools repo. If tools repos enable will fail there is no use of getting the config file.